### PR TITLE
#284: Add support for command `EXPIRETIME`

### PR DESCRIPTION
**Pull request by raknay**
PR to address: Support for command `EXPIRETIME`
Link to [issue](https://github.com/DiceDB/dice/issues/284)

* EXPIRETIME returns the absolute Unix timestamp (since January 1, 1970) in seconds at which the given key will expire
* args should contain only 1 value, the key
* Returns expiration Unix timestamp in seconds.
* Returns -1 if the key exists but has no associated expiration time.
* Returns -2 if the key does not exist.

### CODE REVIEW

File: core/commands.go [hunk 1]
```diff
@@ -125,6 +125,14 @@ var (
 		Arity:    -3,
 		KeySpecs: KeySpecs{BeginIndex: 1, Step: 1},
 	}
+	expiretimeCmdMeta = DiceCmdMeta{
```
Comment (JyotinderSingh) [inline core/commands.go:128]: nit: Please add the metadata for new commands at the bottom of the list, this helps us with certain documentation related automation that we run.

File: core/eval.go [hunk 2]
```diff
@@ -514,6 +514,34 @@ func evalEXPIRE(args []string) []byte {
...
+		return Encode(errors.New("ERR wrong number of arguments for 'expire' command"), false)
+	}
+
+	var key string = args[0]
+
+	obj := Get(key)
+
+	// -2 if key doesn't exist
+	if obj == nil {
+		return RespMinusTwo
```
Comment (JyotinderSingh) [inline core/eval.go:533]: Use RESP_MINUS_2 defined in eval.go

File: core/eval.go [hunk 3]
```diff
@@ -514,6 +514,34 @@ func evalEXPIRE(args []string) []byte {
...
+	}
+
+	exTimeMili, ok := getExpiry(obj)
+	// -1 if key doesn't have expiration time set
+	if !ok {
+		return RespMinusOne
```
Comment (JyotinderSingh) [inline core/eval.go:539]: Use RESP_MINUS_1 defined in eval.go

File: core/eval_test.go [hunk 4]
```diff
@@ -177,6 +178,32 @@ func testEvalGET(t *testing.T) {
...
 }
 
+func testEvalEXPIRETIME(t *testing.T) {
+	tests := map[string]evalTestCase{
+		"key does not exist": {
+			setup:  func() {},
+			input:  []string{"NONEXISTENT_KEY"},
+			output: RespMinusTwo,
+		},
+		"key exists without expary": {
```
Comment (JyotinderSingh) [inline core/eval_test.go:188]:
```suggestion
		"key exists without expiry": {
```

File: core/eval_test.go [hunk 5]
```diff
@@ -177,6 +178,32 @@ func testEvalGET(t *testing.T) {
 	runEvalTests(t, tests, evalGET)
 }
 
+func testEvalEXPIRETIME(t *testing.T) {
```
Comment (JyotinderSingh) [inline core/eval_test.go:181]: Add a test case for the happy path too, where the expiry time is returned.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for adding support for EXPIRETIME @raknay, have left some comments.

**Comment by AshwinKul28** [comment]
Hi @raknay please pull the latest changes from master to pass the linter check

File: core/eval.go (hunk 3 above)
Comment (AshwinKul28) [reply core/eval.go:539]: @JyotinderSingh with the latest linting changes the variable ka changed to RespMinusOne format based on thr golang convention

File: core/commands.go (hunk 1 above)
Comment (raknay) [reply core/commands.go]: Moved to bottom of the list.

File: core/eval.go (hunk 2 above)
Comment (raknay) [reply core/eval.go:534]: We have `RespMinusTwo` defined in `eval.go`.

File: core/eval.go [hunk 6]
```diff
@@ -515,6 +515,34 @@ func evalEXPIRE(args []string) []byte {
...
+	}
+
+	return Encode(int(exTimeMili/1000), true)
```
Comment (JyotinderSingh) [inline core/eval.go:543]:
```suggestion
	return Encode(int(exTimeMili/1000), false)
```

We don't need to encode this as bulk string

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for addressing the reviews @raknay! Just had one small comment.

**Comment by JyotinderSingh** [comment]
> Thanks for addressing the reviews @raknay! Just had one small comment.

Nevermind, I fixed it while merging.

**Comment by JyotinderSingh** [comment]
Thanks for adding this feature @raknay!

**Review by JyotinderSingh — APPROVED** [review]

**Pull request merged**